packages/@react-spectrum/searchfield/src/MobileSearchAutocomplete.tsx

```diff
@@ -0,0 +1,549 @@
+      </Tray>
+    </>
+  );
+});
+
+interface SearchAutocompleteButtonProps extends AriaButtonProps {
+  isQuiet?: boolean,
+  isDisabled?: boolean,
+  isReadOnly?: boolean,
+  isPlaceholder?: boolean,
+  validationState?: ValidationState,
+  inputValue?: string,
+  clearInput?: () => void,
+  children?: ReactNode,
+  style?: React.CSSProperties,
+  className?: string
+}
+
+const SearchAutocompleteButton = React.forwardRef(function SearchAutocompleteButton(props: SearchAutocompleteButtonProps, ref: RefObject<HTMLElement>) {
+  let {
+    isQuiet,
+    isDisabled,
+    isReadOnly,
+    isPlaceholder,
+    validationState,
+    inputValue,
+    clearInput,
+    children,
+    style,
+    className
+} = props;
+  let formatMessage = useMessageFormatter(intlMessages);
+  let valueId = useId();
+  let invalidId = useId();
+  let validationIcon = validationState === 'invalid'
+    ? <AlertMedium id={invalidId} aria-label={formatMessage('invalid')} />
+    : <CheckmarkMedium />;
+
+  let searchIcon = (
+    <Magnifier data-testid="searchicon" />
+  );
+
+  let icon = React.cloneElement(searchIcon, {
+    UNSAFE_className: classNames(
+      textfieldStyles,
+      'spectrum-Textfield-icon'
+    ),
+    size: 'S'
+  });
+
+  let clearButton = (
+    <ClearButton
+      onPress={(e) => {
+        clearInput();
+        props.onPress(e);
+      }}
+      preventFocus
+      aria-label={formatMessage('clear')}
+      excludeFromTabOrder
+      UNSAFE_className={
+        classNames(
+          searchStyles,
+          'spectrum-ClearButton'
+        )
+      }
+      isDisabled={isDisabled} />
+  );
+
+
+  let validation = React.cloneElement(validationIcon, {
+    UNSAFE_className: classNames(
+      textfieldStyles,
+      'spectrum-Textfield-validationIcon',
+      classNames(
+        styles,
+        'spectrum-InputGroup-input-validationIcon'
+      )
+    )
+  });
+
+  let {hoverProps, isHovered} = useHover({});
+  let {buttonProps} = useButton({
+    ...props,
+    'aria-labelledby': [
+      props['aria-labelledby'],
+      props['aria-label'] && !props['aria-labelledby'] ? props.id : null,
+      valueId,
+      validationState === 'invalid' ? invalidId : null
+    ].filter(Boolean).join(' '),
+    elementType: 'div'
+  }, ref);
+
+  return (
+    <FocusRing
+      focusClass={classNames(styles, 'is-focused')}
+      focusRingClass={classNames(styles, 'focus-ring')}>
+      <div
+        {...mergeProps(hoverProps, buttonProps)}
+        aria-haspopup="dialog"
+        ref={ref as RefObject<HTMLDivElement>}
+        style={{...style, outline: 'none'}}
+        className={
+          classNames(
+            styles,
+            'spectrum-InputGroup',
+            {
+              'spectrum-InputGroup--quiet': isQuiet,
+              'is-disabled': isDisabled,
+              'spectrum-InputGroup--invalid': validationState === 'invalid',
+              'is-hovered': isHovered
+            },
+            classNames(
+              comboboxStyles,
```

Comment:
might be better to just duplicate these styles in the searchfield package or move them into spectrum css so you don't need to import the `react-spectrum/combobox` package

Reply:
Good idea, I'll do that.